### PR TITLE
feature: solve the first degree equation

### PR DESCRIPTION
**Pull request by nammon72**
1. What did you do? (mandatory)
- create first-degree-equation.js
- create first-degree-equation.test.js
2. Why would you do that? (mandatory)
- the function use if-else to solve the first-degree equation.
- the test file to test first-degree-equation.test.js
3. An explanation about your `solution or decision or choise` (if needed)

4. Checklist (if needed)

5. Action need to do before merge (if needed)

6. Action need to do after merge (if needed)

8. Others note (if needed)

### CODE REVIEW

File: src/first-degree-equation.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+function solveFirstDegreeEquation(number1, number2) {
+  let result = '';
+  let temp = 0;
+  if (number1 === 0 && number2 === 0) {
+    return result;
```
Comment (anhdiepmmk) [inline src/first-degree-equation.js:5]: should return `the equation has no solution.` instead of empty string

File: src/first-degree-equation.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  } else if (number1 !== 0 && number2 === 0) {
+    result += temp;
```
Comment (anhdiepmmk) [inline src/first-degree-equation.js:7]: return 0;

File: src/first-degree-equation.js (hunk 1 above)
Comment (anhdiepmmk) [inline src/first-degree-equation.js:5]: should return `the equation has infinitely many solutions` instead of empty string

**Review by anhdiepmmk — CHANGES_REQUESTED** [review]

File: src/first-degree-equation.js (hunk 1 above)
Comment (nammon72) [reply src/first-degree-equation.js:5]: so, return a string "the equation has infinitely many solutions", right?

File: src/first-degree-equation.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+function solveFirstDegreeEquation(number1, number2) {
+  let result = '';
+  let temp = 0;
```
[the remark this answers is not in the document]
Comment (nammon72) [reply src/first-degree-equation.js:3]: okay

File: src/first-degree-equation.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  if (number1 === 0 && number2 === 0) {
+    return result;
+  } else if (number1 !== 0 && number2 === 0) {
+    result += temp;
+  } else if (number1 === 0 && number2 !== 0) {
+    return result;
```
[the remark this answers is not in the document]
Comment (nammon72) [reply src/first-degree-equation.js:9]: the result will print "the equation has no solution"?

[the remark this answers is not in the document]
Comment (anhdiepmmk) [reply src/first-degree-equation.js]: yes just return 'the equation has no solution' string

File: src/first-degree-equation.js (hunk 1 above)
Comment (anhdiepmmk) [reply src/first-degree-equation.js]: right

File: src/first-degree-equation.js [hunk 5]
```diff
@@ -1,17 +1,12 @@
 function solveFirstDegreeEquation(number1, number2) {
-  let result = '';
-  let temp = 0;
+  let result = 0;
```
Comment (anhdiepmmk) [inline src/first-degree-equation.js:2]: why need this variable ?

File: src/first-degree-equation.js [hunk 6]
```diff
@@ -1,17 +1,12 @@
...
+    return 0;
   } else if (number1 === 0 && number2 !== 0) {
-    return result;
-  } else {
-    temp = -number2 / number1;
-    result += temp;
-  }
-  return result;
+    return 'The equation has no solutions';
+  } else return -number2 / number1;
```
Comment (anhdiepmmk) [inline src/first-degree-equation.js:9]: else {
  return -number2 / number1; 
}

**Review by anhdiepmmk — CHANGES_REQUESTED** [review]

File: src/first-degree-equation.js (hunk 5 above)
Comment (nammon72) [reply src/first-degree-equation.js:2]: sorry, i forgot to delete

File: src/first-degree-equation.js (hunk 6 above)
Comment (nammon72) [reply src/first-degree-equation.js:9]: thanks for the syntax!

**Review by anhdiepmmk — APPROVED** [review]

**Pull request merged**